src/main/java/mathandel/backend/service/RoleService.java

```diff
@@ -16,63 +19,67 @@
+        User user = userRepository.findById(userId).orElseThrow(() -> new AppException("User doesn't exist"));
 
-        ModeratorRequestStatus moderatorRequestStatus = new ModeratorRequestStatus();
-        moderatorRequestStatus.setName(ModeratorRequestStatusName.PENDING);
+        if(moderatorRequestsRepository.existsByUser(user)) {
+            throw new BadRequestException("Request already submitted");
+        }
 
-        moderatorRequest.setUser(optUser.get())
+        ModeratorRequest moderatorRequest = new ModeratorRequest().setUser(user)
                 .setReason(moderatorRequestTO.getReason())
-                .setModeratorRequestStatus(moderatorRequestStatus);
+                .setModeratorRequestStatus(new ModeratorRequestStatus().setName(ModeratorRequestStatusName.PENDING));
+
         moderatorRequestsRepository.save(moderatorRequest);
 
         return new ApiResponse("Request submitted");
     }
 
-    public List<ModeratorRequestTO> getModeratorRequests() {
-        return moderatorRequestsRepository.findAllByModeratorRequestStatus_Name(ModeratorRequestStatusName.PENDING).stream().map(ServerToClientDataConverter::mapModeratorRequest).collect(Collectors.toList());
+    public Set<ModeratorRequestTO> getModeratorRequests() {
+        return ServerToClientDataConverter.mapModeratorRequests(moderatorRequestsRepository.findAllByModeratorRequestStatus_Name(ModeratorRequestStatusName.PENDING));
     }
 
-
-    public ApiResponse resolveModeratorRequests(List<ModeratorRequestTO> moderatorRequestMessageRequests) {
+    public ApiResponse resolveModeratorRequests(Set<ModeratorRequestTO> resolvedRequests) {
         ModeratorRequest moderatorRequest;
 
-        for (ModeratorRequestTO moderatorRequestMessageRequest : moderatorRequestMessageRequests) {
-            moderatorRequest = moderatorRequestsRepository.findModeratorRequestsByUser_Id(moderatorRequestMessageRequest.getUserId()).orElseThrow(() -> new AppException("No entry in moderator_requests for user " + moderatorRequestMessageRequest.getUserId()));
-            moderatorRequestsRepository.save(moderatorRequest.setModeratorRequestStatus(moderatorRequest.getModeratorRequestStatus().setName(ModeratorRequestStatusName.ACCEPTED)));
+        for (ModeratorRequestTO resolvedRequest : resolvedRequests) {
+            moderatorRequest = moderatorRequestsRepository.findById(resolvedRequest.getId())
+                    .orElseThrow(() -> new AppException("No entry in moderator_requests for user " + resolvedRequest.getUserId()));
+
+            moderatorRequest
+                    .getModeratorRequestStatus()
+                    .setName(resolvedRequest.getModeratorRequestStatus());
+
+            moderatorRequestsRepository.save(moderatorRequest);
         }
 
         return new ApiResponse("Requests resolved");
     }
 
+    // todo sprawic by zwracalo najnowszy request i napisac test
     public ModeratorRequestTO getUserRequests(Long userId) {
```

Comment:
not newest bc user can only have one request which he could later renew if rejected

Reply:
changed to return all requests